mlflow/utils/autologging_utils/safety.py

```diff
@@ -820,6 +919,37 @@ def _validate_new_input(inp):
                 "such exception safe functions / classes.".format(inp)
             )
 
+    def _assert_same_type(autologging_call_input, user_call_input):
+        assert type(autologging_call_input) == type(
+            user_call_input
+        ), "Type of input to original function '{}' does not match expected type '{}'".format(
+            type(autologging_call_input), type(user_call_input)
+        )
+
+    def _assert_same_input_length(autologging_call_input, user_call_input):
```

Comment:
This method name isn't quite accurate. It's asserting that the autologging call input length is `>=` the user call input length to make sure that we didn't delete user arguments.